fdbclient/DatabaseConfiguration.h

```diff
@@ -78,7 +79,17 @@ struct RegionInfo {
 };
 
 struct DatabaseConfiguration {
+	mutable int32_t referenceCount = 1;
```

Comment:
Could we do this by inheriting `ReferenceCounted<DatabaseConfiguration>`, or is there an argument against doing so?

Reply:
IIRC this made forward declarations easier. But I will check again whether `ReferenceCounted` can be used here.